kolibri/core/device/models.py

```diff
@@ -295,3 +295,23 @@ class UserSyncStatus(models.Model):
         is_syncable=False,
     )
     permissions = own | role
+
+
+class OSUser(models.Model):
+    """
+    This class stores a lookup from os username to user id
+    """
+
+    user = models.OneToOneField(
+        FacilityUser,
+        on_delete=models.CASCADE,
+        related_name="os_user",
+        blank=False,
+        null=False,
+        primary_key=True,
+    )
+    # Apparently the max length on linux is 8, on Windows it's 30, and not sure on Mac,
+    # but hopefully 64 should be long enough for anybody!
```

Comment:
For desktop auth on Linux we ended up using both the UID (short numerical identifier) and username (slightly longer human-readable label). One advantage here is Kolibri shows the Kolibri user's username in places, so it's better if that is their name (like `bob`) rather than a funny number like `1000`.

Reply:
Yeah, I thought that a username would be more user readable in general, although, we could just use the uid for connecting the OS user to the FacilityUser, and just use the username for naming the FacilityUser - but it seemed simpler to only have to define one capability for the interface.

Reply:
Ah yes, I think that makes sense indeed. They're both static, after all :)